{
  "id": "CVE-2024-4032",
  "sourceIdentifier": "cna@python.org",
  "published": "2024-06-17T15:15:52.517",
  "lastModified": "2024-06-20T12:44:22.977",
  "vulnStatus": "Awaiting Analysis",
  "descriptions": [
    {
      "lang": "en",
      "value": "The \u201cipaddress\u201d module contained incorrect information about whether certain IPv4 and IPv6 addresses were designated as \u201cglobally reachable\u201d or \u201cprivate\u201d. This affected the is_private and is_global properties of the ipaddress.IPv4Address, ipaddress.IPv4Network, ipaddress.IPv6Address, and ipaddress.IPv6Network classes, where values wouldn\u2019t be returned in accordance with the latest information from the IANA Special-Purpose Address Registries.\n\nCPython 3.12.4 and 3.13.0a6 contain updated information from these registries and thus have the intended behavior."
    },
    {
      "lang": "es",
      "value": "El m\u00f3dulo \"ipaddress\" conten\u00eda informaci\u00f3n incorrecta sobre si ciertas direcciones IPv4 e IPv6 estaban designadas como \"accesibles globalmente\" o \"privadas\". Esto afect\u00f3 las propiedades is_private e is_global de las clases ipaddress.IPv4Address, ipaddress.IPv4Network, ipaddress.IPv6Address y ipaddress.IPv6Network, donde los valores no se devolver\u00edan de acuerdo con la informaci\u00f3n m\u00e1s reciente de los Registros de direcciones de prop\u00f3sito especial de la IANA. CPython 3.12.4 y 3.13.0a6 contienen informaci\u00f3n actualizada de estos registros y, por lo tanto, tienen el comportamiento previsto."
    }
  ],
  "metrics": {},
  "references": [
    {
      "url": "http://www.openwall.com/lists/oss-security/2024/06/17/3",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/22adf29da8d99933ffed8647d3e0726edd16f7f8",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/40d75c2b7f5c67e254d0a025e0f2e2c7ada7f69f",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/895f7e2ac23eff4743143beef0f0c5ac71ea27d3",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/ba431579efdcbaed7a96f2ac4ea0775879a332fb",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/c62c9e518b784fe44432a3f4fc265fb95b651906",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/commit/f86b17ac511e68192ba71f27e752321a3252cee3",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/issues/113171",
      "source": "cna@python.org"
    },
    {
      "url": "https://github.com/python/cpython/pull/113179",
      "source": "cna@python.org"
    },
    {
      "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/NRUHDUS2IV2USIZM2CVMSFL6SCKU3RZA/",
      "source": "cna@python.org"
    },
    {
      "url": "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml",
      "source": "cna@python.org"
    },
    {
      "url": "https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml",
      "source": "cna@python.org"
    }
  ]
}